src/progress_bar.rs

```diff
@@ -592,20 +577,15 @@ impl MultiProgress {
     }
 
     fn push(&self, pos: Option<usize>, pb: ProgressBar) -> ProgressBar {
-        let new = MultiObject {
-            done: false,
-            draw_state: None,
-        };
-
         let mut state = self.state.write().unwrap();
         let idx = match state.free_set.pop() {
             Some(idx) => {
-                state.objects[idx] = Some(new);
+                state.draw_states[idx] = None;
```

Comment:
What exactly is the contract here for `Some` vs `None`? Other than being done, a draw state can now also be `None` if it hasn't been drawn yet? Why hold off on initializing the draw state here?

Reply:
I am not sure what to initialize it to. I can't `draw()` the progressbar yet as that ends up calling back into the `MultiProgress::draw()` which needs the a lock on the `self.state` and assumes that push logic is really done.

I could give it some bogus empty draw state with no lines, but option flt semanticaly cleaner